### ISSUE TITLE
[minor BUG] Success message even on failure. 

### ISSUE THREAD

**Issue by happy-san** [fluttercommunity/flutter_launcher_icons#225]
Version: `0.8.1`

### :speech_balloon: Description

The program prints `✓ Successfully generated launcher icons` even when it fails during execution.

<details>
<summary>Output </summary>

  ════════════════════════════════════════════
            FLUTTER LAUNCHER ICONS (v0.8.0)                    
  ════════════════════════════════════════════
  
• Creating default icons Android

✓ Successfully generated launcher icons
Unhandled exception:
FileSystemException: Cannot open file, path = 'assets/icon/icon.png' (OS Error: No such file or directory, errno = 2)
#0      _File.throwIfError (dart:io/file_impl.dart:635:7)
#1      _File.openSync (dart:io/file_impl.dart:479:5)
#2      _File.readAsBytesSync (dart:io/file_impl.dart:539:18)
#3      createDefaultIcons (package:flutter_launcher_icons/android.dart:35:50)
#4      createIconsFromConfig (package:flutter_launcher_icons/main.dart:99:5)
#5      createIconsFromArguments (package:flutter_launcher_icons/main.dart:58:7)
#6      main (file:///sysroot/home/uprowerp/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_launcher_icons-0.8.1/bin/main.dart:6:3)
#7      _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:299:32)
#8      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:168:12)
pub finished with exit code 255
</details>

### :scroll: Pubspec.yaml

Well, this issue depends on the pubspec file being intentionally wrong.

**Comment by MarkOSullivan94** [comment]
@happy-san can you try with the latest version (v0.9.1) please?

**Comment by happy-san** [comment]
Sure. I'll report back.

**Comment by happy-san** [comment]
@MarkOSullivan94 The bug is still there.

```bash
$ flutter pub run flutter_launcher_icons:main
  ════════════════════════════════════════════
     FLUTTER LAUNCHER ICONS (v0.9.1)                               
  ════════════════════════════════════════════
  
• Creating default icons Android

✓ Successfully generated launcher icons
Unhandled exception:
FileSystemException: Cannot open file, path = 'assets/recipe.pn' (OS Error: No such file or directory, errno = 2)
#0      _File.throwIfError (dart:io/file_impl.dart:635:7)
#1      _File.openSync (dart:io/file_impl.dart:479:5)
#2      _File.readAsBytesSync (dart:io/file_impl.dart:539:18)
#3      decodeImageFile (package:flutter_launcher_icons/utils.dart:35:44)
#4      createDefaultIcons (package:flutter_launcher_icons/android.dart:35:24)
#5      createIconsFromConfig (package:flutter_launcher_icons/main.dart:103:5)
#6      createIconsFromArguments (package:flutter_launcher_icons/main.dart:60:7)
#7      main (file:///sysroot/home/harpreet/lib/flutter/stable/.pub-cache/hosted/pub.dartlang.org/flutter_launcher_icons-0.9.1/bin/main.dart:6:3)
#8      _delayEntrypointInvocation.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:281:32)
#9      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:184:12)
pub finished with exit code 255
```

**Comment by happy-san** [comment]
I think I've found the reason behind this: 

The success message is being `print`ed  in the `finally` block [here](https://github.com/fluttercommunity/flutter_launcher_icons/blob/master/lib/main.dart#L59-#L66)

```dart
    try {
      createIconsFromConfig(yamlConfig);
    } catch (e) {
      stderr.writeln(e);
      exit(2);
    } finally {
      print('\n✓ Successfully generated launcher icons');
    }
```

Following update ought to fix this:

```dart
    try {
      createIconsFromConfig(yamlConfig);
      print('\n✓ Successfully generated launcher icons');
    } catch (e) {
      stderr.writeln(e);
      exit(2);
    } 
```

**Comment by MarkOSullivan94** [comment]
@happy-san feel free to commit a PR to get this merged in 🥳

**Comment by happy-san** [comment]
Should be fixed by #274
